Allow for cross compilation on iPhone on Silicon Macs

RobertRoeb opened the pull request:
Added compile flag --enable-iphonesimulator
  Documented this build instructions

Commit:
Allow for cross compilation on iPhone on Silicon Macs

  Added compile flag --enable-iphonesimulator
  Documented this build instructions

vadz commented on configure.ac:
```diff
@@ -425,6 +425,7 @@ WX_ARG_ONLY_WITH(directfb,      [  --with-directfb         use DirectFB], [wxUSE
 WX_ARG_ONLY_WITH(x11,           [  --with-x11              use X11], [wxUSE_X11="$withval" wxUSE_UNIVERSAL="yes" CACHE_X11=1 TOOLKIT_GIVEN=1])
 WX_ARG_ONLY_WITH(qt,            [  --with-qt               use Qt], [wxUSE_QT="$withval" CACHE_QT=1 TOOLKIT_GIVEN=1])
 WX_ARG_ENABLE(nanox,       [  --enable-nanox          use NanoX], wxUSE_NANOX)
+WX_ARG_ENABLE(iphonesimulator,  [  --enable-iphonesimulator   commpile for iPhone simulator], wxUSE_IPHONESIMULATOR)
```
```suggestion
WX_ARG_ENABLE(iphonesimulator,  [  --enable-iphonesimulator   compile for iPhone simulator], wxUSE_IPHONESIMULATOR)
```

vadz reviewed (commented):
Thanks, I guess it's an improvement but it still doesn't look optimal to specify all of `--with-osx_iphone`, `--enable-iphonesimulator` and `--with-macosx-sdk`. I'm pretty sure we could have `--with-osx_iphonesimulator` replacing the first 2, but I also wonder if we can't either deduce that simulator should be used from the SDK provided (i.e. if it contains `iPhoneSimulator` in its name) or, maybe more logically, choose `iphonesimulator` SDK if the option to build for the simulator is used? I.e. why can't we just make specifying *only* `--with-osx_iphonesimulator` work?

RobertRoeb commented on docs/ios/install.md:
```diff
@@ -25,3 +25,12 @@ The library can also be build via configure/make:
+
+    ../configure --with-osx_iphone --enable-iphonesimulator --enable-monolithic  \
+        --disable-shared --with-macosx-sdk=$(xcrun --sdk iphonesimulator --show-sdk-path)  \
+        --host=aarch64-apple-darwin24.6.0 --build=aarch64-apple-darwin --without-libtiff
```
I don't have an environment variable like "wx" set, so I don't understand how this could find the config.guess script. What did work was this:  
../wxWidgetsRR/configure --with-osx_iphone --enable-iphonesimulator --enable-monolithic  \
        --disable-shared --with-macosx-sdk=$(xcrun --sdk iphonesimulator --show-sdk-path)  \
        --host=$(../wxWidgetsRR/config.guess) --build=aarch64-apple-darwin --without-libtiff
but then I have to write the path twice. I have not found a trick to get the main directory into the options.

RobertRoeb committed:
correct typo

Co-authored-by: VZ <vz-github@zeitlins.org>

Commit:
Added missing message to configure flag

Commit:
Remove old option

RobertRoeb commented:
The title of the last commit is wrong, it now works with just "--with-iphonesimulator" instead of two options

vadz reviewed (commented):
Could you please update `.github/workflows/ci_mac.yml` to use the new option? We really need a way to check that it works (and keeps working). TIA!

RobertRoeb commented:
> We use $WXWIN as env var, sorry. Even if it's not defined, it should be pretty clear what it means.

It took me a little moment to figure that out. For the sake of increasing the chance that people can compile this successfully, I prefer to use paths like this

../wxWidgets/configure --with-osx_iphone --enable-iphonesimulator --enable-monolithic
--disable-shared --with-macosx-sdk=$(xcrun --sdk iphonesimulator --show-sdk-path)
--host=$(../wxWidgets/config.guess) --build=aarch64-apple-darwin --without-libtiff

vadz commented:
> > We use $WXWIN as env var, sorry. Even if it's not defined, it should be pretty clear what it means.
> 
> It took me a little moment to figure that out. For the sake of increasing the chance that people can compile this successfully, I prefer to use paths like this
> 
> ../wxWidgets/configure --with-osx_iphone --enable-iphonesimulator --enable-monolithic --disable-shared --with-macosx-sdk=$(xcrun --sdk iphonesimulator --show-sdk-path) --host=$(../wxWidgets/config.guess) --build=aarch64-apple-darwin --without-libtiff

Sure, this works too. The main idea is to avoid hardcoding the OS version which will change again in a few months.

vadz commented:
> > Could you please update `.github/workflows/ci_mac.yml` to use the new option? We really need a way
> > to check that it works (and keeps working). TIA!
> 
> No idea what that is, but will look.

This is a YAML file describing the steps in (one of) the CI workflow(s), this one building wxiOS. This workflow runs for every PR and push to master and allows to ensure that whatever it is doing doesn't get broken by the future changes.

vadz commented on configure.ac:
```diff
@@ -415,9 +415,11 @@ dnl shouldn't default to wxUSE_ALL_FEATURES
 AC_ARG_WITH(gtk,          [[  --with-gtk[=VERSION]    use GTK+, VERSION can be 3 (default), 2, or "any"]], [wxUSE_GTK="$withval" CACHE_GTK=1 TOOLKIT_GIVEN=1])
 WX_ARG_ONLY_WITH(osx_cocoa,     [  --with-osx_cocoa        use macOS (Cocoa)], [wxUSE_OSX_COCOA="$withval" CACHE_OSX_COCOA=1 TOOLKIT_GIVEN=1])
 WX_ARG_ONLY_WITH(osx_iphone,    [  --with-osx_iphone       use iOS], [wxUSE_OSX_IPHONE="$withval" CACHE_OSX_IPHONE=1 TOOLKIT_GIVEN=1])
+WX_ARG_ONLY_WITH(iphonesimulator, [  --with-osx_iphonesimulator  use iPhone simulator], [wxUSE_OSX_IPHONE="$withval" wxUSE_IPHONESIMULATOR="yes" CACHE_OSX_IPHONE=1 TOOLKIT_GIVEN=1])
```
I think we used to have `--with-carbon` and `--with-cocoa` and then `--with-osx_cocoa` was added to make it more clear what to use for macOS 10 and then the former was removed and `--with-osx` added as it became unambiguous.

Anyhow, they only exist due to history, but IMO this is not a reason to keep duplicating them.

Commit:
Simplify configure further and update docs and github run

  - remove redundant --with-osx_iphonesimulator option in
    favor of just --with_iphonesimulator
  - updated installation to make configure make use of
    config.guess to guess the current host
  - tried to create a github build workflow

RobertRoeb commented:
I added a wxIOSSImmulator build based on guessing parameters and updated configure and install.md. Am I supposed to commit the entire configure?

vadz commented:
> I added a wxIOSSImmulator build based on guessing parameters and updated configure and install.md. Am I supposed to commit the entire configure?

Yes, you have to as long as we keep it in the repository. Please see [the docs](https://github.com/wxWidgets/wxWidgets/blob/master/build/tools/autoconf/README.md) for the recommended way to update it. TIA!

Commit:
Added configure generated by autoconf 2.69 for wxWidgets 3.3.2

Commit:
Removed unnecessary comment about needing up adapt configure command
